packages/api/cms-api/src/user-permissions/auth/user-permissions.guard.ts

```diff
@@ -17,8 +17,11 @@ export class UserPermissionsGuard implements CanActivate {
     ) {}
 
     async canActivate(context: ExecutionContext): Promise<boolean> {
-        const isPublicApi = this.reflector.getAllAndOverride("publicApi", [context.getHandler(), context.getClass()]);
-        if (isPublicApi) {
+        if (this.reflector.getAllAndOverride("disableGlobalGuard", [context.getHandler(), context.getClass()])) {
+            return true;
+        }
+
+        if (this.reflector.getAllAndOverride("publicApi", [context.getHandler(), context.getClass()])) {
```

Comment:
what is the difference between publicApi and disableGlobalGuard?

Reply:
> @PublicApi() can be used to expose a single handler (query, mutation or route) or a whole class (resolver or controller) publicly.

> @DisableGlobalAuthGuard() disables the global auth guard (CometAuthGuard). This may be used if a different authentication method is desired (e.g., basic authentication) for a specific handler or class. It should be used in combination with a custom guard. The custom guard may leverage @PublicApi as well to expose handlers publicly.

https://docs.comet-dxp.com/docs/auth/#available-decorators

Reply:
rtfm :+1:

Reply:
but it's nice to see that some things are documented 🙂